src/App.tsx

```diff
@@ -15,81 +16,160 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-enum SortType {
-  NONE,
-  ALPABET,
-  LENGTH,
-}
-
-type ReorderOptions = {
-  sortType: SortType,
+type State = {
+  goods: string[],
   isReversed: boolean,
+  sortBy: string,
+  resetVisibility: boolean,
+  classVisibility: string,
+  isReverseClicked: boolean,
 };
 
-// Use this function in the render to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+type ButtonsMap = {
+  ALPHABETICALLY: string,
+  BUY_LENGTH: string,
```

Comment:
```suggestion
  BY_LENGTH: string,
```
Please fix this typo in all cases
Btw, it is better to use Enum in this case
https://www.typescriptlang.org/docs/handbook/enums.html#string-enums